test/screens/v2/candidate_generator/on_bus/departures_test.exs

```diff
@@ -0,0 +1,192 @@
+defmodule Screens.V2.CandidateGenerator.Widgets.OnBus.DeparturesTest do
+  use ExUnit.Case, async: true
+
+  alias Screens.Predictions.Prediction
+  alias Screens.Routes.Route
+  alias Screens.Stops.Stop
+  alias Screens.Trips.Trip
+  alias Screens.V2.CandidateGenerator.Widgets.OnBus.Departures
+  alias Screens.V2.Departure
+  alias Screens.V2.ScreenData.QueryParams
+  alias Screens.V2.WidgetInstance.Departures, as: DeparturesWidget
+  alias Screens.V2.WidgetInstance.Departures.NormalSection
+  alias Screens.V2.WidgetInstance.DeparturesNoData
+  alias Screens.V2.WidgetInstance.DeparturesNoService
+  alias ScreensConfig.Screen
+  alias ScreensConfig.V2.OnBus
+
+  import Mox
+  import Screens.Inject
+
+  @stop injected(Stop)
+
+  defp build_config() do
+    %Screen{
+      app_params: %OnBus{
+        evergreen_content: []
+      },
+      vendor: nil,
+      device_id: nil,
+      name: nil,
+      app_id: :on_bus_v2
+    }
+  end
+
+  defp build_departure(
+         route_id,
+         direction_id,
+         route_type \\ :bus,
+         arrival_time \\ ~U[2024-01-01 12:00:00Z]
+       ) do
+    %Departure{
+      prediction: %Prediction{
+        route: %Route{id: route_id, type: route_type},
+        trip: %Trip{direction_id: direction_id},
+        arrival_time: arrival_time
+      }
+    }
+  end
+
+  defp build_stop(id, options \\ %{connecting_ids: [], child_ids: [], parent_id: nil}) do
```

Comment:
This "API" is a bit awkward to use since, if you use any options, you have to pass them all (or else e.g. `options.child_ids` will crash with a nonexistent key error). We could instead have `options \\ %{}` and use `Map.get(options, :child_ids, [])`.

It's also somewhat conventional in Elixir to use keyword lists for these kinds of "options", since there is a special shorthand syntax available when calling a function with a keyword list as the final argument. In this case it would be `options \\ []` and `Keyword.get/3` instead, but otherwise the same.

```elixir
build_stop("100", child_ids: ["200"]) # shorthand keyword-list syntax
```

Reply:
This `options` keyword list feels unnatural to me, but I must admit that it is very clean after converting this function to use it. Thanks for pointing this out, will keep it in mind in the future